samples/custom-bootstrap/README.md

```diff
@@ -2,17 +2,47 @@
 
 This sample creates a simple helloworld service that bootstraps the Envoy proxy with a custom configuration file.
 
-## Starting the service
+## Configuration Options
 
+There are two ways to configure the envoy bootstrap. For both options, it can be useful to reference, [the default bootstrap configuration](/tools/deb/envoy_bootstrap_v2.json) and Envoy's [configuration reference](https://www.envoyproxy.io/docs/envoy/latest/configuration/configuration#config) may be useful
+
+### Configuration Overrides
+
+This option will apply overrides to the default Envoy bootstrap configuration, by merging the two configurations.
+
+For details on how the merging works, see the documentation on the [`--config-yaml`](https://www.envoyproxy.io/docs/envoy/v1.7.1/operations/cli#cmdoption-config-yaml) Envoy flag, and [MergeFrom](https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.message#Message.MergeFrom.details), which describes how this merging takes place.
+
+Note that this option will *not* be processed by templating.
+
+### Fully Custom Configuration
+
+This option will completely overwrite the Envoy bootstrap configuration with your own configuration.
+
+While this provides more flexibility, additional effort may be needed if the default configuration changes.
+
+Note that this option will be processed by templating.
+
+## Running The Example
+
+In this example, we will use a fully custom config and an override. This likely will not be useful in real scenarios, but showcases both methods.
+
+### Creating The Configuration
 First, we need to create a `ConfigMap` resource with our bootstrap configuration.
 
+This file contains two `ConfigMap`s - one for our override, with a filename of `bootstrap_override.json`, and another for our custom config, with a filename of `custom_boostrap.json`.
+
 ```bash
 kubectl apply -f custom-bootstrap.yaml
 ```
 
+### Creating The Service
+
 Next, we can create a service that uses this bootstrap configuration.
 
-To do this, we need to add an annotation, `sidecar.istio.io/bootstrapOverride`, with the name of our ConfigMap as the value.
+To do this, we need to add an annotation with the name of our ConfigMap as the value.
+
+* For a bootstrap override: `sidecar.istio.io/customBootstrapOverride`
```

Comment:
How much do we already rely on the env vars this maps to? There is some value in keeping their names consistent with the names here, but if we can still rename those env vars it may be good to do so:

I'd day that "override" sounds more like replace than merge to me, I suspect at best it's not clear which one it is to most people. Something like *customBootstrapOverride -> bootstrapMerge* and *customBootstrap -> bootstrapReplace* would be a bit clearer.